elastic/kibana
# [Lens] Enables text-based languages (SQL)

## Summary

Closes https://github.com/elastic/kibana/issues/137811

Enables text based query languages in Lens.

![lens](https://user-images.githubusercontent.com/17003240/190333840-cfcca102-42db-4604-b6c5-5402b9714a45.gif)

This is the initial PR so many features are not included. It's primary goal is to set up the architecture of the text based languages in Lens. Things to have in mind:
- You can create only one layer. There are some UI limitations on supporting multiple layers that we are discussing. This is going to work on the future
- Moving from one datasource to another cleanups the unsaved state
- All the data manipulation is now happening from the text based language. For this reason we allow only chart-specific configuration
- You can't create filters by clicking on a chart
- As the data manipulation is done by the query there are some visualizations that might seem "weird" depending on the query.  We want to improve the UX here by adding some warnings in the future.
- Discover to Lens and the opposite direction is not supported for now.
- The suggestions can be improved but this is not on the scope of this PR.

### How to test
The feature is hidden under an advanced setting. To enable it go to advanced settings and switch on the "Enable SQL" setting. If you navigate to Lens and open the dataview picker you will see a new entry

<img width="277" alt="image" src="https://user-images.githubusercontent.com/17003240/191245172-412fc7af-8aad-46cc-ac43-6fbad34e4fcf.png">


### Flaky runners
https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/1303 25 times
https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/1304 100 times

### Checklist
